test/test_mocking_args_matching.sh

```diff
@@ -1,21 +1,25 @@
 #!/bin/bash
 
-test_fails_if_given_arguments_isnt_right() {
-    mock some-command --with-args "one two three"
+test_fails_when_verifying_arguments_even_when_exit_code_is_ignored() {
+    cp -aR ${TEST_ROOT_DIR}/../test-fixtures/mocking-args-strict/* .
 
-    some-command three two one > assertion_output
+    unset RUN_SINGLE_TEST
+    actual=$(${TEST_ROOT_DIR}/../target/sbtest.sh verify_arguments_order)
     assert ${?} failed
 
     expected_error=$(cat <<-EXP
 Unexpected invocation for command 'some-command':
-Got :      <"three two one">
-Expected : <"one two three">
+Got :      <"three trois">
+Expected : <"two deux">
+Unexpected invocation for command 'some-command':
+Got :      <"two deux">
+Expected : <"three trois">
 EXP
 )
-    assert "$(cat assertion_output)" equals "${expected_error}"
+    assert "${actual}" contains "${expected_error}"
```

Comment:
asserting how problems are displayed is an important part of the user experience of this library,  by using contains instead of equals you seem to be hiding something, what is it?

Reply:
The stuff being hidden is caused by the test setup, we essentially have a fake test suite and the output contains much more than the errors themselves. By using contains we can focus on the important part for that test. Otherwise, this test will break when we change, for example, the test report header or footer.

Reply:
The exact test is the following:
```
Running Simple Bash Tests
-------------------------

verify_arguments_order.fails_when_verifying_arguments_even_if_exit_code_is_ignored...FAILED

=========================
FAIL: verify_arguments_order.fails_when_verifying_arguments_even_if_exit_code_is_ignored
-------- STDOUT ---------
Unexpected invocation for command 'some-command':
Got :      <"three trois">
Expected : <"two deux">
Unexpected invocation for command 'some-command':
Got :      <"two deux">
Expected : <"three trois">
-------- STDERR ---------

-------------------------

-------------------------
Ran 1 test

>>> FAILURE (1 problem) <<<
```

Reply:
okay i get the extent of this now, sorry for that.

The test you're replacing is kind of part of the doc on how to use mocking... yes the tests is the doc, so it's important that it's clear and understandable.

Therefore may i ask that the test remains there and that your test becomes a new one at the end of the file ?

And i will agree to the assert contains since this only changes stuff inside the mocking mechanism.

PS. (trailing new lines :) )